juniper_warp/src/lib.rs

```diff
@@ -199,39 +200,45 @@ where
 /// let graphql_filter = make_graphql_filter(schema, context_extractor);
 ///
 /// let graphql_endpoint = warp::path("graphql")
-///     .and(warp::post2())
+///     .and(warp::post())
 ///     .and(graphql_filter);
-/// # }
 /// ```
-pub fn make_graphql_filter<Query, Mutation, Context, S>(
-    schema: juniper::RootNode<'static, Query, Mutation, S>,
+pub fn make_graphql_filter<Query, Mutation, Subscription, Context, S>(
+    schema: juniper::RootNode<'static, Query, Mutation, Subscription, S>,
     context_extractor: BoxedFilter<(Context,)>,
 ) -> BoxedFilter<(warp::http::Response<Vec<u8>>,)>
 where
     S: ScalarValue + Send + Sync + 'static,
-    Context: Send + 'static,
+    Context: Send + Sync + 'static,
     Query: juniper::GraphQLType<S, Context = Context, TypeInfo = ()> + Send + Sync + 'static,
     Mutation: juniper::GraphQLType<S, Context = Context, TypeInfo = ()> + Send + Sync + 'static,
+    Subscription: juniper::GraphQLType<S, Context = Context, TypeInfo = ()> + Send + Sync + 'static,
 {
+    use futures::future::Future as _;
+
     let schema = Arc::new(schema);
     let post_schema = schema.clone();
 
     let handle_post_request =
         move |context: Context, request: GraphQLBatchRequest<S>| -> Response {
             let schema = post_schema.clone();
-            Box::new(
+
+            futures03::compat::Compat01As03::new(
```

Comment:
remove compat

Reply:
[tokio_threadpool](https://crates.io/crates/tokio-threadpool) does not support futures 0.3 yet. 

I would suggest leaving it as is in sync resolvers for now

Reply:
From the docs:

"Note: This crate is deprecated in tokio 0.2.x and has been moved and refactored into various places in the tokio::runtime module of the tokio crate. Note that there is no longer a ThreadPool type, you are instead encouraged to make use of the thread pool used by a Runtime configured to use the threaded scheduler."

Doesn't look like it will be ported. I really don't want futures 01 in here anymore.